### ISSUE TITLE
Wrong repository name on RedHat

### ISSUE THREAD

**Issue by seriv** [puppetlabs/puppetlabs-docker#148]
I get in the output of puppet agent -t:

Error: Execution of '/usr/bin/yum -d 0 -e 0 -y --enablerepo=rhel7-extras install docker-ce' returned 1: Error getting repository data for rhel7-extras, repository not found

Adding 

repo_opt => '--enablerepo=rhel-7-server-extras-rpms'

saves the run.

**Comment by davejrt** [comment]
What version of RHEL are you running?

**Comment by nikdoof** [comment]
The repo name may be correct if you've manually specified it in your yum.conf, but the RHSM delivered name for the extras repo is ```rhel-7-server-extras-rpms```

```
[root@monitoring ~]# subscription-manager repos --list
+----------------------------------------------------------+
    Available Repositories in /etc/yum.repos.d/redhat.repo
+----------------------------------------------------------+

...

Repo ID:   rhel-7-server-extras-rpms
Repo Name: Red Hat Enterprise Linux 7 Server - Extras (RPMs)
Repo URL:  https://cdn.redhat.com/content/dist/rhel/server/7/7Server/$basearch/extras/os
Enabled:   0
```

**Comment by thatsk** [comment]
https://github.com/puppetlabs/puppetlabs-docker/issues/161

You can try this.

**Comment by davejrt** [comment]
@nikdoof thanks for clarifying that. I'm assuming you also have this issue, but are manually working around it at this point? I'm just canvassing for community feedback because I'm surprised this hasn't been brought up more in the past. I can't see anything in the garethr module that we forked that relates to this.

**Comment by nikdoof** [comment]
@davejrt Yes I guess people are working around the issue, if you're using a local Satellite/Katello instance, mirror or just the CentOS7 repos I guess its not an issue as you're expecting the extras repository to be named something different, but if you're direct feeding off the RHCDN then the repo name is different. Ideally it should be using the out of the box config for RHEL

Here are similar issues reported: [#1](https://github.com/garethr/garethr-docker/issues/462#issuecomment-242556048), [#2](https://github.com/garethr/garethr-docker/issues/737), [#3](https://github.com/garethr/garethr-docker/issues/248).

TL;DR: The garethr version solution was to add repo_opt in the first place, rather than correct the default repo name for RHEL7 to the RHSM ID.

Now that i've got some free time i'll actually submit a PR for this :)
